[Yuan Wei] iP

tyw2811 opened the pull request:
# Duke by yuan wei
>The ~~best~~ **bestest** version of Duke??


What is Duke?
- A command line chat bot.
- A tool to manages your task list.

Whats good about Duke?
1. Text based!
2. Easy to learn!
2. Fuss free setup! Just download from this [link](https://github.com/tyw2811/ip/releases/tag/A-Jar).

Features of Duke:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
#
If you are a Java programmer, here is a snippet of the `main` method:
```java
public static void main(String[] args) {
    new Duke("./tasks.txt").run();
}
```
Ok have fun with Duke!! :slightly_smiling_face:

Commit:
Add Gradle support

tyw2811 committed:
Implemented Greet, Echo, Exit

tyw2811 committed:
Implemented Add, List

tyw2811 committed:
Added Task class, implemented Mark/Unmark of tasks

tyw2811 committed:
Implemented ToDo, Event, Deadlines

tyw2811 committed:
Edited some testing cases

tyw2811 committed:
Refactored Duke.java and added comments

tyw2811 committed:
Added Exceptions to handle errors

tyw2811 committed:
Implemented Delete

tyw2811 committed:
Update JavaDocs

tyw2811 committed:
Add FileReader to read and save from Task.txt

tyw2811 committed:
Add Date and Time to tasks

tyw2811 committed:
Merge branch 'branch-Level-7'

tyw2811 committed:
Merge branch 'branch-Level-8'

tyw2811 committed:
Refactor code to extract out related code

tyw2811 committed:
Divide classes into packages and subpackages

tyw2811 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Add gradle support

tyw2811 committed:
Add JUnit tests for Deadline and EmptyTaskException

tyw2811 committed:
Add Javadoc to the JUnit test classes

tyw2811 committed:
Add find command, refactor OutputCommands into separate classes

tyw2811 committed:
Merge branch 'branch-A-JavaDoc'

tyw2811 committed:
Merge branch 'branch-Level-9'

tyw2811 committed:
Add CheckStyle to gradle, fix code to match CheckStyle

tyw2811 committed:
Add basic javafx from javafx tutorial 1

JustinS00 reviewed (commented):
LTGM. Good documentations.

jeromehjj commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,61 @@
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui) throws InvalidTaskNumberException {
+        if (index.equals("") || !isInteger(index) || (Integer.parseInt(index) - 1) < 0
+                || (Integer.parseInt(index) - 1) >= taskList.size()) {
```
Perhaps you might want to implement a function to check whether the index is valid? 
Eg. `isValidIndex(taskList)` to avoid overcomplicating the expression highlighted.

jeromehjj commented on src/test/java/duke/exception/EmptyTaskExceptionTest.java:
```diff
@@ -0,0 +1,21 @@
+package duke.exception;
+
+import duke.task.Deadline;
+import org.junit.jupiter.api.Test;
```
I think using line breaks here would make a clearer distinction between your import statements

jeromehjj reviewed (commented):
LGTM! I think your code is well-written in terms of documentation and follows the coding standards very well. In terms of code quality, I think the naming aspect was very well done when reviewing your code. Just some small suggestions regarding having complicated expressions and the ordering of import statements. Good job! 👍

tyw2811 committed:
Implement JavaFX GUI, remove text based UI

tyw2811 committed:
Merge branch 'branch-Level-10'

tin-jy commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,41 @@
+/**
+ * A class for the todo command.
+ */
+public class TodoCommand extends Command {
```
Perhaps the three Task commands can be further grouped

tin-jy commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+     * @return The string representation of the Deadline object to be stored in the file.
+     */
+    @Override
+    public String toFile() {
```
Interesting to have a different toString() and toFile() function

tin-jy commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,90 @@
+     *
+     * @return A string indicating if the Task is done.
+     */
+    protected String getStatusIcon() {
```
I like this to simplify toString()

tin-jy commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,90 @@
+     *
+     * @return true if the task has changed state.
+     */
+    public boolean setUnDone() {
```
Not sure if this is correct capitalization

tin-jy reviewed (commented):
Nice!

tyw2811 committed:
Refactor task.txt location for jar file

tyw2811 committed:
Add assert to ensure certain assumptions are met

There are certain assumptions that should hold true, otherwise the
code will run into unintended errors.

Adding these assertions will ensure that if there are errors thrown
due to these assumptions being violated, the location of the error and
the reason will be clear.

tyw2811 committed:
Improve on code quality throughout

tyw2811 committed:
Merge pull request #1 from tyw2811/branch-A-Assertions

Add assert to ensure certain assumptions are met

tyw2811 committed:
Merge branch 'branch-A-CodeQuality'

tyw2811 committed:
Add fixed duration tasks

tyw2811 committed:
Add meow command

tyw2811 committed:
Refactor Duke to CatBot

tyw2811 committed:
Update README

tyw2811 committed:
Update README.md

tyw2811 committed:
Update README.md

tyw2811 committed:
Refactor tasks.txt location

tyw2811 committed:
Update ReadMe.md

tyw2811 committed:
Update ReadMe.md

tyw2811 committed:
Implement help command

tyw2811 committed:
Update README.md
